packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -638,8 +667,35 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
       e: React.SyntheticEvent,
       searchInputProps: DropdownSearchInputProps,
     ) => {
-      if (keyboardKey.getCode(e) === keyboardKey.Tab && !_.isNil(highlightedIndex)) {
-        selectItemAtIndex(highlightedIndex)
+      switch (keyboardKey.getCode(e)) {
+        case keyboardKey.Tab:
+          if (!_.isNil(highlightedIndex)) {
+            selectItemAtIndex(highlightedIndex)
+          }
+          break
+        case rtl ? keyboardKey.ArrowRight : keyboardKey.ArrowLeft:
```

Comment:
For readability, I'd rather leave the switch case as a simple condition and extract the body of a case in a private method:

something like:

```typescript
private setActiveIndexToLastSelectedItem = () => {
  const { value } = this.state as { value: ShorthandValue[] }
  if (this.props.multiple && value.length > 0) {
    this.trySetState({ activeIndex: value.length - 1 })
  }
}

// in the switch/case
switch (keyboardKey.getCode(e)) {
  // ...
  case keyboardKey.ArrowLeft:
    if (!rtl) {
      this.setActiveIndexToLastSelectedItem()
    }
    break
  case keyboardKey.ArrowRight:
    if (rtl) {
      this.setActiveIndexToLastSelectedItem()
    }
    break
  // ...
}
```
what do you think?

Reply:
we will refactor this in the future, with behaviours, so any improvement made now will only get removed then. right now I think this is good enough.

Reply:
ok, maybe I've changed my mind.